File: backend/src/ShaktiUdyog.Domain/Entities/Invoice.cs
using ShaktiUdyog.Domain.Constants;

namespace ShaktiUdyog.Domain.Entities;

public class Invoice
{
    public Guid Id { get; set; }
    public required string InvoiceNumber { get; set; }
    public Guid CompanyId { get; set; }
    public Company Company { get; set; } = null!;
    public Guid? OrderId { get; set; }
    public Order? Order { get; set; }
    public DateTimeOffset IssueDateUtc { get; set; }
    public DateTimeOffset? DueDateUtc { get; set; }
    public decimal Subtotal { get; set; }
    public decimal Tax { get; set; }
    public decimal Discount { get; set; }
    public decimal Freight { get; set; }
    public decimal Packing { get; set; }
    public decimal OtherCharges { get; set; }
    public decimal Total { get; set; }
    public decimal AmountPaid { get; set; }
    public decimal BalanceDue { get; set; }
    public string Currency { get; set; } = "INR";
    public string Status { get; set; } = InvoiceStatuses.Draft;
    public string? PaymentTerms { get; set; }
    public string? Notes { get; set; }
    public string? HsnSacCode { get; set; }
    public Guid? DocumentId { get; set; }
    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;
    public byte[] RowVersion { get; set; } = [];

    public List<InvoiceItem> Items { get; set; } = [];
    public List<InvoiceStatusHistory> StatusHistory { get; set; } = [];
    public List<InvoiceAttachment> Attachments { get; set; } = [];
}

File: frontend/src/types/invoice.ts
/**
 * Invoicing and Payment domain models.
 */

export interface InvoiceItem {
  id: string;
  description: string;
  hsnSacCode: string | null;
  quantity: number;
  unit: string;
  unitPrice: number;
  taxPercent: number;
  lineTotal: number;
}

export interface Payment {
  id: string;
  paymentReference: string;
  method: string;
  amount: number;
  paymentDateUtc: string;
  status: string;
  createdAtUtc: string;
}

export interface InvoiceListItem {
  id: string;
  orderId: string;
  invoiceNumber: string;
  orderNumber: string | null;
  issueDateUtc: string;
  dueDateUtc: string | null;
  total: number;
  amountPaid: number;
  balanceDue: number;
  currency: string;
  status: string;
  companyName: string | null;
  companyLogoUrl: string | null;
  companyEmail: string | null;
  companyPhone: string | null;
}

export interface InvoiceDetail extends InvoiceListItem {
  subtotal: number;
  tax: number;
  documentId: string | null;
  items: InvoiceItem[];
  payments: Payment[];
}
